exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java

```diff
@@ -0,0 +1,152 @@
+
+  private final TraceServiceGrpc.TraceServiceBlockingStub blockingStub;
+  private final ManagedChannel managedChannel;
+  private final long deadline;
+
+  /**
+   * Creates a new Jaeger gRPC Span Reporter with the given name, using the given channel.
+   *
+   * @param channel the channel to use when communicating with the Jaeger Collector.
+   * @param deadline max waiting time for the collector to process each span batch. When set to 0 or
+   *     to a negative value, the exporter will wait indefinitely.
+   */
+  private OtlpGrpcSpanExporter(ManagedChannel channel, long deadline) {
+    this.managedChannel = channel;
+    this.blockingStub = TraceServiceGrpc.newBlockingStub(channel);
+    this.deadline = deadline;
+  }
+
+  /**
+   * Submits all the given spans in a single batch to the Jaeger collector.
+   *
+   * @param spans the list of sampled Spans to be exported.
+   * @return the result of the operation
+   */
+  @Override
+  public ResultCode export(List<SpanData> spans) {
+    ExportTraceServiceRequest exportTraceServiceRequest =
+        ExportTraceServiceRequest.newBuilder()
+            .addAllResourceSpans(SpanAdapter.toProtoResourceSpans(spans))
+            .build();
+
+    try {
+      TraceServiceGrpc.TraceServiceBlockingStub stub = this.blockingStub;
+      if (deadline > 0) {
+        stub = stub.withDeadlineAfter(deadline, TimeUnit.MILLISECONDS);
+      }
+
+      // for now, there's nothing to check in the response object
+      // noinspection ResultOfMethodCallIgnored
+      stub.export(exportTraceServiceRequest);
+      return ResultCode.SUCCESS;
+    } catch (StatusRuntimeException e) {
+      // Retryable codes from https://github.com/open-telemetry/oteps/pull/65
+      switch (e.getStatus().getCode()) {
+        case CANCELLED:
+        case DEADLINE_EXCEEDED:
+        case RESOURCE_EXHAUSTED:
+        case OUT_OF_RANGE:
+        case UNAVAILABLE:
+        case DATA_LOSS:
+          return ResultCode.FAILED_RETRYABLE;
+        default:
+          return ResultCode.FAILED_NOT_RETRYABLE;
+      }
+    } catch (Throwable t) {
+      return ResultCode.FAILED_NOT_RETRYABLE;
+    }
+  }
+
+  /**
+   * Creates a new builder instance.
+   *
+   * @return a new instance builder for this exporter
+   */
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /**
+   * Initiates an orderly shutdown in which preexisting calls continue but new calls are immediately
+   * cancelled. The channel is forcefully closed after a timeout.
+   */
+  @Override
+  public void shutdown() {
+    try {
+      managedChannel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      logger.log(Level.WARNING, "Failed to shutdown the gRPC channel", e);
+    }
+  }
+
+  /** Builder utility for this exporter. */
+  public static class Builder {
+    private ManagedChannel channel;
+    private long deadline = 1_000; // ms
```

Comment:
rather than the comment, name the variable "deadlineMillis"

Reply:
Done.